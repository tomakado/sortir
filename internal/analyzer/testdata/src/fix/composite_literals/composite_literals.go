package composite_literals

type Person struct {
	Name    string
	Age     int
	Country string
	City    string
	Street  string
}

func main() {
	// Multi-line struct literal with unsorted fields
	p1 := &Person{
		Street:  "Main St", // want `composite literal elements are not sorted`
		City:    "New York",
		Country: "USA",
		Age:     30,
		Name:    "John",
	}

	// Another multi-line struct literal
	p2 := Person{
		Country: "Canada", // want `composite literal elements are not sorted`
		City:    "Toronto",
		Street:  "Queen St",
		Name:    "Jane",
		Age:     25,
	}

	// Single-line struct literal (should remain single-line)
	p3 := Person{Country: "UK", City: "London", Age: 35, Name: "Bob", Street: "Baker St"} // want `composite literal elements are not sorted`

	// Map literal with multi-line format
	m := map[string]int{
		"zebra": 1, // want `composite literal elements are not sorted`
		"apple": 2,
		"mango": 3,
		"banana": 4,
	}

	_, _, _, _ = p1, p2, p3, m
}