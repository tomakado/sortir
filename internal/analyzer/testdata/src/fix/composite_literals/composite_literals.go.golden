package composite_literals

type Person struct {
	Age     int
	City    string
	Country string
	Name    string
	Street  string
}

func main() {
	// Multi-line struct literal with unsorted fields
	p1 := &Person{
		Age:     30,
		City:    "New York",
		Country: "USA",
		Name:    "John",
		Street:  "Main St", // want `composite literal elements are not sorted`
	}

	// Another multi-line struct literal
	p2 := Person{
		Age:     25,
		City:    "Toronto",
		Country: "Canada", // want `composite literal elements are not sorted`
		Name:    "Jane",
		Street:  "Queen St",
	}

	// Single-line struct literal (should remain single-line)
	p3 := Person{Age: 35, City: "London", Country: "UK", Name: "Bob", Street: "Baker St"} // want `composite literal elements are not sorted`

	// Map literal with multi-line format
	m := map[string]int{
		"apple": 2,
		"banana": 4,
		"mango": 3,
		"zebra": 1, // want `composite literal elements are not sorted`
	}

	_, _, _, _ = p1, p2, p3, m
}